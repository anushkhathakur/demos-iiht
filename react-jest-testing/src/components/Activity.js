import React from 'react';

const STATUS = {
    NORMAL : 'normal',
    HOVER : 'hover'
}
export default class Activity extends React.Component{
    constructor(props){
        super(props);

        this.state= {
            cssClass : STATUS.NORMAL
        }
    }
    onMouseEnter(){
        this.setState({ cssClass : STATUS.HOVER });
    }
    onMouseLeave(){
        this.setState({ cssClass: STATUS.NORMAL });
    }

    // render the component
    render(){
        return(
            <a className={this.state.cssClass}
                href={this.props.url || '#'}
                onMouseEnter={this.onMouseEnter.bind(this)}
                onMouseLeave={this.onMouseLeave.bind(this)}
                >
                {this.props.children}
            </a>
        );
    }
}