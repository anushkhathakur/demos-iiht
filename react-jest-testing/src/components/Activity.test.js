import React from 'react';
import Activity from './Activity';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer'

// snapshot testing
test('Href changes the style on hover',()=>{
    //const component = render(<Activity></Activity>);
    const component = renderer.create(<Activity></Activity>);

    let _elements = component.toJSON();
    expect(_elements).toMatchSnapshot();

    // trigger events 
    _elements.props.onMouseEnter();
    // re-rendering
    _elements = component.toJSON();
    expect(_elements).toMatchSnapshot();
});

test('should have href value and css class as hover',()=>{
    const component = renderer.create(<Activity url="http://localhost"></Activity>);

    let _elements = component.toJSON();
    expect(_elements).toMatchSnapshot();

    // trigger events 
    _elements.props.onMouseEnter();
    // re-rendering
    _elements = component.toJSON();
    expect(_elements).toMatchSnapshot();

    // trigger event
    _elements.props.onMouseLeave();
    // re-rendering
    _elements = component.toJSON();
    expect(_elements).toMatchSnapshot(); 
});