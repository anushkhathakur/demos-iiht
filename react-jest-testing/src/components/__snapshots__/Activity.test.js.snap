// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Href changes the style on hover 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`Href changes the style on hover 2`] = `
<a
  className="hover"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`should have href value and css class as hover 1`] = `
<a
  className="normal"
  href="http://localhost"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`should have href value and css class as hover 2`] = `
<a
  className="hover"
  href="http://localhost"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;

exports[`should have href value and css class as hover 3`] = `
<a
  className="normal"
  href="http://localhost"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
